### PR TITLE
Jenkins: get rid of macOS warnings

### PR DESCRIPTION
**Pull request by stefantalpalaru**

### CODE REVIEW

**Comment by stefantalpalaru** [comment]
Fixes https://github.com/status-im/nimbus-eth2/issues/3076

File: Jenkinsfile [hunk 1]
```diff
@@ -21,14 +21,15 @@ def runStages(nodeDir) {
 			}
 
 			cache(maxCacheSize: 250, caches: [
-				[$class: "ArbitraryFileCache", excludes: "", includes: "**/*", path: "${WORKSPACE}/${nodeDir}/vendor/nimbus-build-system/vendor/Nim/bin"],
```
Comment (jakubgs) [inline Jenkinsfile:24]: Do we want to build the Nim compiler every time?

File: Makefile [hunk 2]
```diff
@@ -31,9 +31,17 @@ CPU_LIMIT := 0
...
+  CPU_LIMIT_CMD :=
 else
-	CPU_LIMIT_CMD := cpulimit --limit=$(CPU_LIMIT) --foreground --
+  CPU_LIMIT_CMD := cpulimit --limit=$(CPU_LIMIT) --foreground --
+endif
+
+# TODO: move this to nimbus-build-system
+ifeq ($(shell uname), Darwin)
+  # Scary warnings in large volume: https://github.com/status-im/nimbus-eth2/issues/3076
+  SILENCE_WARNINGS := 2>&1 | grep -v "failed to insert symbol" | grep -v "could not find object file symbol for symbol" || true
```
Comment (jakubgs) [inline Makefile:42]: You can avoid invoking two instances of `grep` by using `-e` twice for the same call.

**Comment by jakubgs** [comment]
It would be good to explain in the commit(s) or the PR description why these "scary" warnings aren't actually scary and can be ignored.

File: Jenkinsfile (hunk 1 above)
Comment (stefantalpalaru) [reply Jenkinsfile:24]: Yes. It's 1-2 minutes of run time, in exchange for a cleaner setup that doesn't have to deal with old libraries and object files from previous runs.

File: Makefile (hunk 2 above)
Comment (stefantalpalaru) [reply Makefile:42]: Nice one, but not very readable. How many people know what `-e` does without reading the man page?

**Comment by stefantalpalaru** [comment]
We're linking object files with debug info with ones without it. Perfectly justified, from our point of view, but `dsymutil` complains verbosely about it, with no switch to silence it.

File: Jenkinsfile (hunk 1 above)
Comment (jakubgs) [reply Jenkinsfile:24]: That's no that bad.

File: Makefile (hunk 2 above)
Comment (jakubgs) [reply Makefile:42]: Really? Your worry is if someone knows `-e` in this spaghetti of `Makefile`s?

**Comment by jakubgs** [comment]
Okay, then lets include that explanation in the description.

**Comment by stefantalpalaru** [comment]
Included in the commit desc: https://github.com/status-im/nimbus-eth2/commit/555ce310cc85aba794951a740665b1d0685abea1

File: Makefile (hunk 2 above)
Comment (stefantalpalaru) [reply Makefile:42]: :-)

**Pull request merged**